share/commands/package_upgrade.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/python
+import sys
+import json
+
+class Section(object):
+# Non optimal, it seems to evaluate 2 times each key, I do not know why.
+
+  def __init__(self, oldJson, newJson):
+    self.json = oldJson
+    self.newJson = newJson
+    self.oldMap = {}
+    self.newMap = {}
+    self.mapping = {
+        'PACKAGE_MANAGER_ALLOW_UNTRUSTED': self.manager_untrusted,
+        'PACKAGE_ARCHITECTURE_SPECIFIC': self.architecture_specific,
+        'PACKAGE_POST_HOOK_COMMAND': self.hook_upgrade,
+        'PACKAGE_VERSION_SPECIFIC': self.version_specific_upgrade,
+        'PACKAGE_MANAGER_OPTIONS': self.manager_options,
+        'PACKAGE_ARCHITECTURE': self.architecture,
+        'PACKAGE_VERSION': self.version_upgrade,
+        'PACKAGE_MANAGER': self.manager_upgrade,
+        'PACKAGE_STATE': self.state_upgrade,
+        'PACKAGE_LIST': self.package_upgrade
+      }
+
+# Acces to the oldMap via apt or rpm
+  def parameter_traduction(self, keyList):
+      for iKey in keyList:
+          if iKey in self.oldMap:
+              return self.oldMap[iKey]
+      return ""
+
+# Mapping functions to upgrade parameter values
+  def hook_upgrade(self):
+    if 'RPM_PACKAGE_POST_HOOK_COMMAND' in self.oldMap:
+        return self.parameter_traduction(['RPM_PACKAGE_POST_HOOK_COMMAND'])
+    else: return ""
+
+  def state_upgrade(self):
```

Comment:
It would have been more readable to put those functions in the same order as the self.mapping structure

Reply:
And since almost all those functions are one liner, put them directly in the structure